### PR TITLE
Add latam spanish headers

### PR DESCRIPTION
**Pull request by JuanoD**
Add missing latam spanish keymaps

## Description

Latam keyboards are different from spanish keyboards. Add its own layout to make configuration easy for latins.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/keymap_extras/keymap_spanish_latam.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 QMK
```
Comment (zvecr) [inline quantum/keymap_extras/keymap_spanish_latam.h:1]: Implementing directly is invalid. Content is now configured within `data/constants/keycodes/extras/` and generated

**Review by zvecr — CHANGES_REQUESTED** [review]

File: quantum/keymap_extras/keymap_spanish_latam.h (hunk 1 above)
Comment (JuanoD) [reply quantum/keymap_extras/keymap_spanish_latam.h:1]: Done

File: data/constants/keycodes/extras/keycodes_spanish_latam_0.0.1.hjson [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+        }
+        "S(LA_MINS)": {
+            "key": "LA_UNDS",
+            "label": "_",
+        }
+/* AltGr symbols
+ * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───────┐
+ * │ ¬ │   │   │   │   │   │   │   │   │   │   │ \ │   │       │
+ * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─────┤
+ * │     │ @ │   │ € │   │   │   │   │   │   │   │   │ ~ │     │
```
Comment (fauxpark) [inline data/constants/keycodes/extras/keycodes_spanish_latam_0.0.1.hjson:309]: AltGr+E does not produce € on Windows 10 (22H2).

File: docs/ja/reference_keymap_extras.md [hunk 3]
```diff
@@ -61,6 +61,7 @@ HID Keyboard/Keypad usage ページでは 256 未満の usage が定義されて
 | Slovenian                   | `keymap_slovenian.h`             |
 | Spanish                     | `keymap_spanish.h`               |
 | Spanish (Dvorak)            | `keymap_spanish_dvorak.h`        |
+| Spanish (Latam)             | `keymap_spanish_latam.h`         |
```
Comment (fauxpark) [inline docs/ja/reference_keymap_extras.md:64]:
```suggestion
```

File: docs/reference_keymap_extras.md [hunk 4]
```diff
@@ -67,6 +67,7 @@ These headers are located in [`quantum/keymap_extras/`](https://github.com/qmk/q
 |Slovenian                        |`keymap_slovenian.h`             |`sendstring_slovenian.h`            |
 |Spanish                          |`keymap_spanish.h`               |`sendstring_spanish.h`              |
 |Spanish (Dvorak)                 |`keymap_spanish_dvorak.h`        |`sendstring_spanish_dvorak.h`       |
+|Spanish (Latam)                  |`keymap_spanish_latam.h`         |`sendstring_spanish_latam.h`        |
```
Comment (fauxpark) [inline docs/reference_keymap_extras.md:70]:
```suggestion
|Spanish (Latin America)          |`keymap_spanish_latin_america.h` |`sendstring_spanish_latin_america.h`|
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: quantum/keymap_extras/sendstring_spanish_latin_america.h [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+    // BS    TAB      LF       VT       FF       CR       SO       SI
+    KC_BSPC, KC_TAB,  KC_ENT,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+    // DLE   DC1      DC2      DC3      DC4      NAK      SYN      ETB
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+    // CAN   EM       SUB      ESC      FS       GS       RS       US
+    XXXXXXX, XXXXXXX, XXXXXXX, KC_ESC,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+
+    //       !        "        #        $        %        &        '
+    KC_SPC,  LA_1,    LA_2,    LA_3,    LA_4,    LA_5,    LA_6,    LA_QUOT,
```
Comment (fauxpark) [inline quantum/keymap_extras/sendstring_spanish_latin_america.h:97]: This file needs adjusting too.

File: docs/feature_macros.md [hunk 6]
```diff
@@ -69,9 +69,9 @@ The current list of available languages is:
 | **italian_osx_iso** | **jis** | **latvian** | **lithuanian_azerty** |
 | **lithuanian_qwerty** | **norman** | **norwegian** | **portuguese** |
 | **portuguese_osx_iso** | **romanian** | **serbian_latin** | **slovak** |
-| **slovenian** | **spanish_dvorak** | **spanish** | **swedish** |
-| **turkish_f** | **turkish_q** | **uk** | **us_international** |
-| **workman** | **workman_zxcvm** |
+| **slovenian** | **spanish_dvorak** | **spanish_latam** | **spanish** |
```
Comment (fauxpark) [inline docs/feature_macros.md:72]:
```suggestion
| **slovenian** | **spanish_dvorak** | **spanish_latin_america** | **spanish** |
```

File: quantum/keymap_extras/keymap_spanish_latin_america.h
Comment (fauxpark) [inline quantum/keymap_extras/keymap_spanish_latin_america.h:1]: Needs to be regenerated.

**Comment by JuanoD** [comment]
Would you prefer if I squash the commits?

**Review by fauxpark — APPROVED** [review]

**Pull request merged**